addons/binding/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/handler/BrokerHandler.java

```diff
@@ -179,40 +178,40 @@ protected void assignSSLContextProvider(BrokerHandlerConfig config, MqttBrokerCo
      */
     protected MqttBrokerConnection createBrokerConnection() throws IllegalArgumentException {
         String host = config.host;
-        if (StringUtils.isBlank(host) || host == null) {
+        if ((host == null) || StringUtils.isBlank(host)) {
             throw new IllegalArgumentException("Host is empty!");
         }
 
-        final MqttBrokerConnection connection = new MqttBrokerConnection(host, config.port, config.secure,
-                config.clientID);
+        MqttBrokerConnection connection = new MqttBrokerConnection(host, config.port, config.secure, config.clientID);
+        connection.setQos(config.qos);
+        connection.setRetain(config.retainMessages);
 
-        final String username = config.username;
-        final String password = config.password;
-        if (StringUtils.isNotBlank(username) && password != null) {
-            connection.setCredentials(username, password); // Empty passwords are allowed
+        if (StringUtils.isNotBlank(config.username) && (config.password != null)) {
+            connection.setCredentials(config.username, config.password); // Empty passwords are allowed
         }
 
-        final String topic = config.lwtTopic;
+        String topic = config.lwtTopic;
```

Comment:
You are not assigning anything to `topic` are you? So why removing the `final`?

Reply:
Well, 
![grafik](https://user-images.githubusercontent.com/19878123/52504283-41ed7980-2be8-11e9-85ef-5146b356a7cb.png)
Would you guys agree about "final"?

Reply:
Yup, I know we maintainers have different opinions regarding this detail. That's why I approve new bindings that do not follow this style. But here you are changing existing code :)